packages/world-modules/src/modules/keyswithvalue/KeysWithValueModule.sol

```diff
@@ -45,16 +46,24 @@ contract KeysWithValueModule is Module {
 
     // Extract source table id from args
     ResourceId sourceTableId = ResourceId.wrap(abi.decode(args, (bytes32)));
-    ResourceId targetTableSelector = getTargetTableId(MODULE_NAMESPACE, sourceTableId);
+    ResourceId targetTableId = getTargetTableId(MODULE_NAMESPACE, sourceTableId);
 
     IBaseWorld world = IBaseWorld(_world());
 
+    // Register the target namespace if it doesn't exist yet
+    if (!ResourceIds.getExists(targetTableId.getNamespaceId())) {
+      (bool registrationSuccess, bytes memory registrationReturnData) = address(world).delegatecall(
+        abi.encodeCall(world.registerNamespace, (targetTableId.getNamespaceId()))
+      );
+      if (!registrationSuccess) revertWithBytes(registrationReturnData);
```

Comment:
I think we already played around with this, but is it a bad idea to wrap this logic in a free function like

```solidity
function delegateCallOrRevert(address target, bytes memory callData) returns (bytes memory) {
  (bool success, bytes memory returnData) = target.delegateCall(callData);
  if (!success) revertWithBytes(returnData);
  return returnData;
}

delegateCallOrRevert(address(world), abi.encodeCall(world.registerNamespace, (targetTableId.getNamespaceId())));
```

Reply:
Sadly every time memory is passed around (even in free functions) it increases gas usage. Might still be worth for code organization purposes and to make sure handle the `success` flag, but would move this to a followup (including exploring the gas impact)